bin/nbb

```diff
@@ -344,6 +339,24 @@ find_centos_7_ami_ids() {
   done
 }
 
+shutdown_instances() {
+  aws ec2 describe-instances --filters "Name=instance-state-name,Values=running" | jq -c -r '.Reservations[] .Instances[] .InstanceId' > instance_list
+  echo "You are about to terminate the following instances:"
+  cat instance_list
+  read -r -p "Are you sure? [y/N] " response
+  if [[ $response =~ ^([yY][eE][sS]|[yY])$ ]]
+  then
+    IFS=$'\n'  # make newlines the separator
+    for i in `cat ./instance_list`; do
+      echo "Shutting down $i"
+      aws ec2 terminate-instances --instance-ids $i
```

Comment:
@sethlindberg I'm sure they're relatively quick in most cases as you pointed out; my concern is more around questions like, "what if I have 100 instances to terminate!?" 😧 

If this is _blocking_, we'll find the user stuck in that loop for quite some time.

Now, _is that actually bad?_ In this case there might be a good reason to go fully synchronous here; if we can't do anything else with our teardown until all these instances are gone anyway, maybe we just have to bite the bullet and wait on them one-by-one. However, this creates another problem: if somehow one of these calls times out and the `aws` command exits status `1`, _will it force an `exit 1` of the entire orchestration script, all the way up the stack?_

> The only situation where this is different is in instance termination, where it shuts things down right away and then gives a four hour grace period to catch a possible mistake or bad actor.

I wanted to point this out actually; the use case here is to prevent `make destroy` from barfing up at us due to AWS dependency violations. One thing I've seen pretty often is that an instance still exists inside a subnet so the entire `make destroy` process explodes and you don't know where you stand anymore. It's rather frustrating indeed.

The problem is that we _DO_ need to **terminate** these instances before we can be sure not to get thrown said `DepdendencyViolation` errors. AFAIK a simple change of state to `shutdown` isn't enough to do it. I've seen instances in that state in my test account derail `make destroy` in the past (plus I think the user's still billed for a `shutdown` instance, right?)

Anyway, I wanted to point out the consequences of making this synchronous. If I'm in the minority opinion here, hey no worries! :smiley: I just wanted to be sure it was at least discussed in some fashion before glossing it over, that's all.